src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/TypeFetcher.cs

```diff
@@ -154,77 +129,82 @@ public Type LoadTypeFromCrefValues(IList<string> crefValues)
 
-            // Attempt to load type from friendly name, search through all loaded assemblies.
-            if (contractType == null)
-            {
-                var potentialTypes = new List<Type>();
+            var loadedAssemblies = AppDomain.CurrentDomain.GetAssemblies()
+                .Where(a => !a.IsDynamic);
 
-                var loadedAssemblies = AppDomain.CurrentDomain.GetAssemblies()
-                    .Where(a => !a.IsDynamic);
+            foreach (var loadedAssembly in loadedAssemblies)
+            {
+                var contractType = loadedAssembly.GetType(typeName);
 
-                foreach (var loadedAssembly in loadedAssemblies)
+                if (contractType == null)
                 {
-                    try
-                    {
-                        var potentialTypesInAssembly = loadedAssembly.GetTypes()
-                            .Where(t => t.Name.Equals(typeName) || t.FullName.Equals(typeName));
-                        potentialTypes.AddRange(potentialTypesInAssembly);
-                    }
-                    catch (ReflectionTypeLoadException e)
-                    {
-                        // Unable to load types from assembly due to missing dependencies.
-                        // The types that are available, however, are included in the exception.
-                        // Check to see if a potential type is specified
-                        foreach (var loadedType in e.Types)
-                        {
-                            if (loadedType != null && (loadedType.Name.Equals(typeName)
-                                || loadedType.FullName.Equals(typeName)))
-                            {
-                                potentialTypes.Add(loadedType);
-                            }
-                        }
-                    }
+                    continue;
                 }
 
-                if (potentialTypes.Count > 1)
-                {
-                    var errorMessage = string.Format(SpecificationGenerationMessages.CannotUniquelyIdentifyType,
-                        typeName, string.Join(" ", potentialTypes.Select(type => type.FullName)));
+                _typeMap.Add(typeName, contractType);
+
+                return contractType;
+            }
 
-                    throw new TypeLoadException(errorMessage);
-                }
 
-                contractType = potentialTypes.FirstOrDefault();
+            var errorMessage = string.Format(
+                SpecificationGenerationMessages.TypeNotFound,
+                typeName,
+                string.Join(" ", _contractAssemblyPaths.Select(Path.GetFileName)));
+
+            throw new TypeLoadException(errorMessage);
+        }
+
+        /// <summary>
+        /// Gets the type from the cref value.
+        /// </summary>
+        /// <param name="crefValues">The list of cref values.</param>
+        /// <returns>The type.</returns>
+        public Type LoadTypeFromCrefValues(IList<string> crefValues)
```

Comment:
>public [](start = 8, length = 6)

Does this need to be public?

Reply:
other custom filters can leverage this so public is fine 

---
In reply to: [216434864](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/145#discussion_r216434864) [](ancestors = 216434864)